src/sortStudents.ts

```diff
@@ -1,16 +1,75 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedOrder = order === 'asc';
+  const studentsCopy = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case SortType.Name:
+      studentsCopy.sort((a, b) => (sortedOrder
+        ? a.name.localeCompare(b.name)
+        : b.name.localeCompare(a.name)
+      ));
+      break;
+
+    case SortType.Surname:
+      studentsCopy.sort((a, b) => (sortedOrder
+        ? a.surname.localeCompare(b.surname)
+        : b.surname.localeCompare(a.surname)
+      ));
+      break;
```

Comment:
If you update your enam it allows you make this
```suggestion
    case SortType.Name:
    case SortType.Surname:
      studentsCopy.sort((a, b) => (sortedOrder
        ? a[sortBy].localeCompare(b[sortBy])
        : b[sortBy].surname.localeCompare(a[sortBy])
      ));
      break;
```